### PR TITLE
Fix parser for aachen

### PR DESCRIPTION
**Pull request by Y0hy0h**
The menu structure changed for the canteens in Aachen.

Before: https://web.archive.org/web/20170807225026/http://www.studierendenwerk-aachen.de/speiseplaene/academica-w.html  
After: http://www.studierendenwerk-aachen.de/speiseplaene/academica-w.html

This broke the parser, which is sad. ;)

### CODE REVIEW

**Comment by Y0hy0h** [comment]
Disclaimer: I rebased to change the order of the last two commits, but GitHub sorts them by date which messes up the hierarchy. I did this because all commits except 3f56bc6 ("Fix parsing of new menu for aachen") are refactorings to the old code.

File: aachen.py [hunk 1]
```diff
@@ -1,79 +1,118 @@
...
+
+
+def parse_description(description_container, legend):
+    name_parts = []
+    notes = set()
+    for element in description_container:
+        if type(element) is Tag and element.name == 'sup':
+            notes.update(element.text.strip().split(','))
+        else:
+            name_parts.append(element.string.strip())
```
Comment (azrdev) [inline aachen.py]: calling `parser.parse("aachen", "vita", None))` this crashes here for me, end of backtrace:
~~~
  File "openmensa-parsers/aachen.py", line 104, in parse_description
    name_parts.append(element.strip())
TypeError: 'NoneType' object is not callable

**Comment by Y0hy0h** [comment]
The Aachen website is a complete mess:

https://web.archive.org/web/20171130094808/http://www.studierendenwerk-aachen.de/speiseplaene/academica-w.html  
If you look at "Dienstag, 28.11.2017", you will see "Tellergericht" and "Vegetarisch" have empty duplicates.

https://web.archive.org/web/20171130094833/http://www.studierendenwerk-aachen.de/speiseplaene/vita-w.html  
Notice here that "Hauptbeilagen" and "Nebenbeilage" miss the category names.

I decided to ignore all entries, where the category or the name is missing. For the Academica, this behaves as I want and removes the empty duplicates. For the Vita, however, there will be no "Hauptbeilagen" as long as the website does not name them, which is the workaround I would propose.

**Comment by azrdev** [comment]
> The Aachen website is a complete mess:

Given their [current rearrangements](http://www.studierendenwerk-aachen.de/de/aktuelles/beitrag/umstellung-der-online-speiseplaene.html) I'd propose to postpone your structured rewrite until the site has become somewhat stable, and go with the quick-/minimal-fix #60 instead (But, as I said over there, s/o else would have to decide. Wait for it :)

**Comment by Y0hy0h** [comment]
Choose the one you find more maintainable. I don't mind if you like the other more. :)

Note also that #60 adds parsing of the nutritional values, adding them to the meal's notes. My PR does not do that.

**Comment by Y0hy0h** [comment]
Once we have a version of the website without glaring issues, I will take a snapshot using the [WaybackMachine](https://archive.org/web/) which should allow us to write a regression test. This will prevent the parser from missing essential features, like just happended with the notes missing.

**Comment by Y0hy0h** [comment]
I just created #63 which implements a regression test, like stated in the comment above.

**Comment by mswart** [comment]
Sorry for the late response, I was on a long vacation.

@Y0hy0h Many thanks for your PR. I have merged your changes but combined them into one commit to keep the history short.
I added a check to make the menu optional for days (like last Monday, 1th of January): cdf1b725675a3c2bc2dc6b68f5cc76c9d750eabf.

@Y0hy0h Do you want to function as maintainer for the parser "Aachen"? I do not use any canteens from Aachen and am short on time: help would be very useful.

**Comment by Y0hy0h** [comment]
As stated in the [issue comment](https://github.com/mswart/openmensa-parsers/issues/61#issuecomment-355923862) I am willing to maintain the Aachener parsers. :)

**Pull request merged**